new.go

```diff
@@ -6,62 +6,40 @@ import (
 	"fmt"
 	"io"
 	"os"
+	"time"
 
 	v1 "github.com/google/go-containerregistry/pkg/v1"
 	"github.com/google/go-containerregistry/pkg/v1/empty"
 	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/types"
 
 	"github.com/buildpacks/imgutil/layer"
 )
 
-func NewCNBImage(repoName string, store ImageStore, options ImageOptions) (*CNBImageCore, error) {
+func NewCNBImage(options ImageOptions) (*CNBImageCore, error) {
 	image := &CNBImageCore{
-		Image: options.BaseImage,
-		Store: store,
-		// required
-		repoName: repoName,
-		// optional
-		preferredMediaTypes: options.MediaTypes,
+		Image:               options.BaseImage, // the working image
+		createdAt:           getCreatedAt(options),
+		preferredMediaTypes: GetPreferredMediaTypes(options),
 		preserveHistory:     options.PreserveHistory,
 		previousImage:       options.PreviousImage,
 	}
 
+	// ensure base image
 	var err error
 	if image.Image == nil {
-		image.Image, err = emptyV1(options.Platform)
+		image.Image, err = emptyV1(options.Platform, image.preferredMediaTypes)
 		if err != nil {
 			return nil, err
 		}
 	}
-	if image.Image, err = OverrideMediaTypes(image.Image, options.MediaTypes); err != nil {
-		return nil, err
-	}
 
+	// ensure windows
 	if err = prepareNewWindowsImageIfNeeded(image); err != nil {
 		return nil, err
 	}
 
-	createdAt := NormalizedDateTime
-	if !options.CreatedAt.IsZero() {
-		createdAt = options.CreatedAt
-	}
-	if err = image.MutateConfigFile(func(c *v1.ConfigFile) {
-		c.Created = v1.Time{Time: createdAt}
-		c.Container = ""
-	}); err != nil {
-		return nil, err
-	}
-
-	if !options.PreserveHistory {
-		if err = image.MutateConfigFile(func(c *v1.ConfigFile) {
-			for j := range c.History {
-				c.History[j] = v1.History{Created: v1.Time{Time: createdAt}}
-			}
-		}); err != nil {
-			return nil, err
-		}
-	}
```

Comment:
Moved this out of new method to avoid mutating the digest when there is a base image